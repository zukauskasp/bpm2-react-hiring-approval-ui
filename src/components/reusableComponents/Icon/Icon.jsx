import React from "react";
import PropTypes from "prop-types";

const Icon = ({ icon, color, ...otherProps }) => {

    return (
        <img alt="" src={process.env.PUBLIC_URL + '/assets/icons/' + icon + '/' + color + '.svg'} {...otherProps} />
    )
}

Icon.propTypes = {
    icon: PropTypes.oneOf(['add', 'alert', 'back', 'check',
        'collapse', 'duplicate', 'edit', 'expand', 'info', 'next',
        'remove', 'removeInCircle', 'search', 'trash', 'upload'
    ]),
    color: PropTypes.oneOf(['white', 'dark']),
    otherProps: PropTypes.any,
}

Icon.defaultProps = {
    color: "dark"
}


export default Icon;