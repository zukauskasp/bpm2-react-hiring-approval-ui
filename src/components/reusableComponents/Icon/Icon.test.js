import React from 'react';
import { shallow, mount } from 'enzyme';
import Icon from "./Icon";

describe('Icon', () => {

    const props = {
        icon: "add",
        color: "dark"
    }

    afterAll(() => {
        mount.cleanUp();
      });

    it('expect to match latest snapshot', () => {
        const wrapper = mount(<Icon {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('expect to render Icon component', () => {
        expect(shallow(<Icon {...props} />).length).toEqual(1);
    });

    it('expect to have src generated from the props', () => {
        const wrapper = shallow(<Icon {...props} />);
        expect(wrapper.prop('src')).toEqual(process.env.PUBLIC_URL + '/assets/icons/' + props.icon + '/' + props.color + '.svg');
    });

    it('expect to render default color dark', () => {
        const wrapper = shallow(<Icon icon={props.icon} />);
        expect(wrapper.prop('src')).toEqual(process.env.PUBLIC_URL + '/assets/icons/' + props.icon + '/dark.svg');
    });

    it('expect to render other props correctly', () => {
        const wrapper = shallow(<Icon icon={props.icon} onClick={() => { }} />);
        expect(wrapper.prop('nonExistingProp')).toBeUndefined();
        expect(typeof wrapper.prop('onClick')).toEqual('function');
    });

    it('expect to mock event correctly', () => {
        const mockCallBack = jest.fn();
        const wrapper = shallow(<Icon icon={props.icon} onClick={mockCallBack} />);
        wrapper.simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

});