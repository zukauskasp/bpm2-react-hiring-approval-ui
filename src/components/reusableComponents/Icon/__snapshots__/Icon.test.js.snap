// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon expect to match latest snapshot 1`] = `
"<Icon icon=\\"add\\" color=\\"dark\\">
  <img alt=\\"\\" src=\\"/assets/icons/add/dark.svg\\" />
</Icon>"
`;
